tests/test_warmingup_unit_cases.py

```diff
@@ -72,6 +124,42 @@ def test_3a(self):
         # Just a "problem is not infeasible"
         heat_problem = run_optimization_problem(HeatProblem, base_folder=base_folder)
 
-        demand_matching_test(heat_problem, heat_problem.extract_results())
-        energy_conservation_test(heat_problem, heat_problem.extract_results())
-        heat_to_discharge_test(heat_problem, heat_problem.extract_results())
+        results = heat_problem.extract_results()
+        parameters = heat_problem.parameters(0)
+        bounds = heat_problem.bounds()
+
+        demand_matching_test(heat_problem, results)
+        energy_conservation_test(heat_problem, results)
+        heat_to_discharge_test(heat_problem, results)
+
+        # We only check the flow directions for the time-steps that there is flow in the pipe.
+        inds = np.round(1 - results["Pipe_e53a__is_disconnected"]).astype(bool)
+        np.testing.assert_allclose(
+            np.round(results["Pipe_e53a__flow_direct_var"][inds]) * 2.0 - 1.0,
+            np.sign(results["Pipe_e53a.Q"][inds]),
+        )
+
+        for buffer in heat_problem.heat_network_components.get("buffer", []):
+            np.testing.assert_allclose(
+                results[f"{buffer}.Heat_buffer"], results[f"{buffer}.Heat_flow"]
+            )
+            np.testing.assert_allclose(
+                results[f"{buffer}.HeatIn.Heat"] - results[f"{buffer}.HeatOut.Heat"],
+                results[f"{buffer}.Heat_buffer"],
```

Comment:
shouldn't we add this test to the utils_tests? we do have it in that list for the heat_demand check